src/parser/bison_parser.y

```diff
@@ -1070,6 +1086,49 @@ date_literal:
 		}
 	;
 
+interval_literal:
+		int_literal duration_field { $$ = Expr::makeIntervalLiteral($1->ival, $2); delete $1; }
+	|   INTERVAL STRING opt_datetime_field {
```

Comment:
I would split the logic here und use
`|   INTERVAL STRING datetime_field
| INTERVAL STRING`